lib/rust/ensogl/component/grid-view/src/selectable.rs

```diff
@@ -17,6 +19,69 @@ pub mod highlight;
 
 
 
+// ======================
+// === MovementTarget ===
+// ======================
+
+/// An internal structure describing where selection would go after being moved (i.e.
+/// after navigating with arrows). If moving the selection would put it outside the grid, the
+/// [`OutOfBounds`] variant contains the direction in which the grid boundary would be crossed.
+/// Otherwise, the [`Location`] variant contains row and column in the grid.
+#[derive(Copy, Clone, Debug)]
+enum MovementTarget {
+    Location { row: Row, col: Col },
+    OutOfBounds(frp::io::keyboard::ArrowDirection),
+}
+
+impl MovementTarget {
+    /// Calculate row and column of the nearest entry in given direction from given row and col.
+    /// Returns a [`MovementTarget::Location`] if the entry is in bounds of a grid with given
+    /// amount of rows and columns. Returns [`MovementTarget::OutOfBounds`] otherwise.
+    fn nearest_in_direction(
```

Comment:
For some reason "nearest" sounds strange to me. Maybe it suggests there is some space between selected and returned entry? What about "next_in_direction"?